Algorithm/QCAlgorithm.cs

```diff
@@ -1442,7 +1442,7 @@ public Security AddSecurity(SecurityType securityType, string symbol, Resolution
                 }
 
                 Symbol symbolObject;
-                if (!SymbolCache.TryGetSymbol(symbol, out symbolObject))
+                if (!SymbolCache.TryGetSymbol(symbol, out symbolObject) || symbolObject.ID.Market != market)
```

Comment:
While here please rename symbol variable to `ticker` and add security type check to future proof it. Tiny chance of 2 colliding USA ticker names.